pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -299,13 +312,99 @@ func (e *TokensController) syncSecret() {
 				retry = retriable
 			}
 		default:
-			// Update token if needed
-			if retriable, err := e.generateTokenIfNeeded(sa, secret); err != nil {
-				klog.Errorf("error populating serviceaccount token %s/%s for service account %s: %v", secretInfo.namespace, secretInfo.name, secretInfo.saName, err)
-				retry = retriable
+			// Check if the legacy service account token can be cleaned up
+			if e.shouldCleanToken(secret, sa) {
+				// Delete token
+				klog.V(4).Infof("syncSecret(%s/%s), auto-generated token has not been used for more than one year, deleting token", secretInfo.namespace, secretInfo.name)
+				if retriable, err := e.deleteToken(secretInfo.namespace, secretInfo.name, secretInfo.uid); err != nil {
+					klog.Errorf("error deleting auto-generated serviceaccount token %s/%s for service account %s: %v", secretInfo.namespace, secretInfo.name, secretInfo.saName, err)
+					retry = retriable
+				}
+			} else {
+				// Update token if needed
+			  if retriable, err := e.generateTokenIfNeeded(sa, secret); err != nil {
+					klog.Errorf("error populating serviceaccount token %s/%s for service account %s: %v", secretInfo.namespace, secretInfo.name, secretInfo.saName, err)
+					retry = retriable
+			  }
+			}
+		}
+	}
+}
+
+func (e *TokensController) shouldCleanToken(secret *v1.Secret, sa *v1.ServiceAccount) bool {
+
+	if !e.legacyTokenCleanUp {
+		return false
+	}
+
+	// determine whether the secret is referenced by the service account, if not referenced, the secret is manually generated and should not be cleaned up
```

Comment:
the order should be:1
1. check the tracking time 
2. check secret is referenced by the sa in case the tracking label is mutated by others
3. check mounted or not.

Reply:
ignore the comment. i thought only auto generated legacy tokens are labeled.